src/huntsman/pocs/observatory.py

```diff
@@ -775,13 +785,17 @@ def _take_flat_observation(self, exptimes, observation, fits_headers=None, dark=
             camera_events[cam_name] = {'event': camera_event, 'filename': filename}
 
         # Block until done exposing on all cameras
-        while not all([info['event'].is_set() for info in camera_events.values()]):
-            self.logger.debug('Waiting for flat-field images...')
-            time.sleep(1)
-
+        timeout = max(exptimes.values()).to_value(u.second) + flat_field_timeout
+        self.logger.debug(f"Waiting for flat-fields with timeout of {timeout}.")
+        if not wait_for_events(camera_events, timeout=timeout, sleep_delay=1):
+            self.logger.error("Timeout while waiting for flat fields.")
+
+        # Remove camera_events that timed out, removing them from the remaining flat-fielding
+        camera_events = {cam_name: event for cam_name, event in camera_events.items(
+                         ) if event.is_set()}
```

Comment:
```suggestion
        camera_events = {
            cam_name: event 
            for cam_name, event 
            in camera_events.items() 
            if event.is_set()
        }
```

Reply:
Ewwww I don't like this multi-line style very much!